File: sample-event-emitter/src/test/java/nl/haploid/octowight/TestData.java
package nl.haploid.octowight;

import nl.haploid.octowight.repository.AtomChangeEventDmo;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static final String ATOM_LOCUS = "sunnydale";

	public static final String ATOM_TYPE = "spike";

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static AtomChangeEventDmo atomChangeEventDmo() {
		final AtomChangeEventDmo event = new AtomChangeEventDmo();
		event.setAtomId(nextLong());
		event.setAtomLocus(ATOM_LOCUS);
		event.setAtomType(ATOM_TYPE);
		return event;
	}

	public static String topic() {
		return UUID.randomUUID().toString();
	}
}

File: sample-event-emitter/src/test/scala/nl/haploid/octowight/sample/TestData.scala
package nl.haploid.octowight.sample

import java.util.{Random, UUID}

import nl.haploid.octowight.sample.repository.AtomChangeEventDmo

object TestData {
  val AtomLocus = nextString
  val AtomCategory = nextString

  def nextLong: Long = new Random().nextLong

  // TODO: must be a nicer way to say that id is unused
  def atomChangeEventDmo =
    new AtomChangeEventDmo(id = 666, atomId = nextLong, atomOrigin = AtomLocus, atomCategory = AtomCategory)

  def topic = nextString

  def nextString = UUID.randomUUID.toString
}
